File: VisualStudio/VBasic/wgpframe/clsPfJointLoad.vb
'
' Copyright (c)2016 S C Harrison
' Refer to License.txt for terms and conditions of use.
'
Option Explicit On

Public Class clsPfJointLoad

#Const isUsingExcel = 0

    Public key As Integer

    Public jt As Byte
    Public fx As Double          '.. horizontal load @ a joint ..
    Public fy As Double          '.. vertical   load @ a joint ..
    Public mz As Double          '.. moment applied  @ a joint ..

    Sub initialise()
        key = 0
        jt = 0
        fx = 0
        fy = 0
        mz = 0
    End Sub

    Sub setValues(ByVal LoadKey As Integer, ByVal Node As Integer, ByVal ForceX As Double, ByVal ForceY As Double, ByVal Moment As Double)
        key = LoadKey
        jt = Node
        fx = ForceX
        fy = ForceY
        mz = Moment
    End Sub

    Function sprint() As String
        sprint = StrLPad(Format(key, "##0"), 8) _
                 & StrLPad(Format(jt, "##0"), 6) _
                 & StrLPad(Format(fx, "0.000"), 15) _
                 & StrLPad(Format(fy, "0.000"), 15) _
                 & StrLPad(Format(mz, "0.000"), 15)
    End Function

    Sub cprint()
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & sprint
    End Sub

    Sub fprint(ByVal fp As Integer)
        Print(fp, sprint)
    End Sub


#If isUsingExcel = 1 Then

    Sub wbkPrint(ByVal dataRec As Range, ByVal RecNum As Byte)
        With dataRec
            .Offset(RecNum, 0).Value = key
            .Offset(RecNum, 1).Value = jt
            .Offset(RecNum, 2).Value = fx
            .Offset(RecNum, 3).Value = fy
            .Offset(RecNum, 4).Value = mz
        End With
    End Sub

    Sub wbkRead(ByVal dataRec As Range, ByVal RecNum As Byte)
        With dataRec
            key = .Offset(RecNum, 0).Value
            jt = .Offset(RecNum, 1).Value
            fx = .Offset(RecNum, 2).Value
            fy = .Offset(RecNum, 3).Value
            mz = .Offset(RecNum, 4).Value
        End With
    End Sub

#End If

End Class

File: Eclipse/w7cpframe/src/clsPfJointLoad.java
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
import java.io.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;


    class clsPfJointLoad
    {


        public int key = 0;

        public int jt = 0;
        public double fx = 0;          //.. horizontal load @ a joint ..
        public double fy = 0;          //.. vertical   load @ a joint ..
        public double mz = 0;          //.. moment applied  @ a joint ..



        public void initialise()
        {
            key = 0;
            jt = 0;
            fx = 0;
            fy = 0;
            mz = 0;
        }

        public void setValues(int LoadKey, int Node, double ForceX, double ForceY, double Moment)
        {
            key = LoadKey;
            jt = Node;
            fx = ForceX;
            fy = ForceY;
            mz = Moment;
        }

        public String sprint()
        {
            String s;

            s = "";
            s = s + String.format("%8d",key);
            s = s + String.format("%6d",jt);
            s = s + String.format("%15.4f",fx);
            s = s + String.format("%15.4f",fy);
            s = s + String.format("%15.4f",mz);

            return s;
        }

        public void cprint()
        {
            System.out.println(sprint());
        }

        public void fprint(BufferedWriter fp)
        {
            try {
				fp.write(sprint());
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
        }

        public void fgetData(BufferedReader fp)
        {
            String s="";

            System.out.println("fgetData ...");

            try {
				s = fp.readLine();
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
            System.out.println(s);
            sgetData(s);

            System.out.println("... fgetData");
        }

        public void sgetData(String s)
        {
            String[] dataflds = new String[10]; //(0 To 9);
            int i;

            System.out.println("sgetData ...");

            System.out.println(s);

            //Trim Trailing Spaces from input String
            String regExpTrimPattern = "^\\s+|\\s+$"; //trim trailing spaces          
            Pattern regExpPatternObj = Pattern.compile(regExpTrimPattern);
            Matcher regExpMatches = regExpPatternObj.matcher(s);
            String regResult = regExpMatches.replaceAll("");
            
            //Get Data fields from input String
            String regExpNumberPattern = "-?\\d+(?:[,.]\\d+)?";
            regExpPatternObj = Pattern.compile(regExpNumberPattern);
            regExpMatches = regExpPatternObj.matcher(regResult);
            i=0;
            while (regExpMatches.find()){
            	dataflds[i] = regExpMatches.group().trim();
            	i = i+1;
            }

            key = Integer.parseInt(dataflds[0]);
            jt = Integer.parseInt(dataflds[1]);
            fx = Double.parseDouble(dataflds[2]);
            fy = Double.parseDouble(dataflds[3]);
            mz = Double.parseDouble(dataflds[4]);


            System.out.println("... sgetData");
        }




    } //class